### PR TITLE
Update some metric types, fix some wording (#12122)

### PR DESCRIPTION
**Pull request by pmmukh**
- This PR merges in changes from #12122, along with other docs changes not in stable-website

### CODE REVIEW

File: website/content/docs/internals/telemetry.mdx [hunk 1]
```diff
@@ -93,8 +93,9 @@ These metrics represent operational aspects of the running Vault instance.
...
-| `vault.core.mount_table.size`                       | Size of a particular mount table. This metric is labeled by table type (auth or logical) and whether or not the table is replicated (local or not)                                                                                                                                                                                                                                                                                          | objects      | summary |
+| `vault.core.expiration_time_epoch`                  | Time as epoch (seconds since Jan 1 1970) at which license will expire.                                                                                                                                                                                                                                                                                                                                                                      | sm           | gauge  |
```
Comment (ncabatoff) [inline website/content/docs/internals/telemetry.mdx:96]: This shouldn't be (expiration_time_epoch), it's a 1.8 thing.

File: website/content/docs/internals/telemetry.mdx [hunk 2]
```diff
@@ -375,54 +376,72 @@ These metrics relate to the supported [storage backends][storage-backends].
...
+| `vault.raft-storage.transaction`                                       | Time to insert operations into a single log.                                                                                                                                                                      | ms                                | timer   |
+| `vault.raft-storage.entry_size`                                        | The total size of a Raft entry during log application in bytes.                                                                                                                                                   | bytes                             | summary  |
+| `vault.raft_storage.bolt.freelist.`<br/>`free_pages`                          | Number of free pages in the freelist.                                                                                                                                                                      | pages                             | gauge  |
```
Comment (ncabatoff) [inline website/content/docs/internals/telemetry.mdx:427]: Ditto the bolt metrics.

Comment (pmmukh) [reply website/content/docs/internals/telemetry.mdx:427]: Aha! That's why they weren't published, gotcha, thanks!!

**Review by ncabatoff — APPROVED** [review]

**Comment by mladlow** [comment]
@pmmukh in the future you should be able to put the docs-cherrypick label on the original PR and skip the step of creating a separate PR.

**Comment by pmmukh** [comment]
@mladlow thanks! will be sure to do the label addition in the future, I totally missed it yesterday until after I'd already merged in the PR 😅

**Pull request merged**